From 86916069ddeb5ab1a5d33b70c43050f615fe6eb4 Mon Sep 17 00:00:00 2001
From: David Holsgrove <david.holsgrove@petalogix.com>
Date: Thu, 26 Jan 2012 10:20:19 +1000
Subject: [PATCH 11/41] microblaze: dl-machine.h update function elf_machine_rela

prevents;

do-rel.h:148:11: error: too many arguments to function 'elf_machine_rela'

Signed-off-by: David Holsgrove <david.holsgrove@petalogix.com>
---
 sysdeps/microblaze/dl-machine.h |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/sysdeps/microblaze/dl-machine.h b/sysdeps/microblaze/dl-machine.h
index 5bd11bb..8fdd1c3 100644
--- a/sysdeps/microblaze/dl-machine.h
+++ b/sysdeps/microblaze/dl-machine.h
@@ -163,6 +163,8 @@ asm (TRAMPOLINE_TEMPLATE (_dl_runtime_resolve, fixup) \
      ".set _dl_runtime_profile, _dl_runtime_resolve");
 #endif
 
+/* The PLT uses Elf32_Rela relocs.  */
+#define elf_machine_relplt elf_machine_rela
 
 /* Mask identifying addresses reserved for the user program,
    where the dynamic linker should not map anything.  */
@@ -291,10 +293,10 @@ elf_machine_plt_value (struct link_map *map, const Elf32_Rela *reloc,
 auto inline void __attribute__ ((always_inline))
 elf_machine_rela (struct link_map *map, const Elf32_Rela *reloc,
 		  const Elf32_Sym *sym, const struct r_found_version *version,
-		  void *const reloc_addr_arg)
+		  void *const reloc_addr_arg, int skip_ifunc)
 {
   Elf32_Addr *const reloc_addr = reloc_addr_arg;
-  const unsigned int r_type = ELF32_R_TYPE (reloc->r_info);
+  const int r_type = ELF32_R_TYPE (reloc->r_info);
 
   if (__builtin_expect (r_type == R_MICROBLAZE_64_PCREL, 0))
     PUT_REL_64(reloc_addr, map->l_addr + reloc->r_addend);
@@ -369,7 +371,8 @@ elf_machine_rela_relative (Elf32_Addr l_addr, const Elf32_Rela *reloc,
 
 auto inline void
 elf_machine_lazy_rel (struct link_map *map,
-		      Elf32_Addr l_addr, const Elf32_Rela *reloc)
+		      Elf32_Addr l_addr, const Elf32_Rela *reloc,
+		      int skip_ifunc)
 {
   Elf32_Addr *const reloc_addr = (void *) (l_addr + reloc->r_offset);
   if (ELF32_R_TYPE (reloc->r_info) == R_MICROBLAZE_JUMP_SLOT)
-- 
1.7.1

